# Filtering DE_Tdu_Tpr_for_Tms
# Input DE_Tdu_Tpr_min10_3rep.txt; only those orthologs showing non-biased mapping will be remained
# 5,400 non-biased orthologs are in file bayes_flag_sig_Filtered_Tdu-Tpr_overlap.V2.csv

In [1]:
import pandas as pd
import sys
import os
import glob

In [2]:
os.chdir("/ufrc/soltis/shan158538/TragFL_NewAnalysis/OutPut/DE_analysis/Tdu_Tpr_voom_Tms/")

In [3]:
df = pd.read_table("DE_Tdu_Tpr_min10_3rep.txt")

In [4]:
df.head()

logFC    ...            B
Tpr_TRINITY_DN11257_c2_g1|Tdu_TRINITY_DN16696_c... -0.053919    ...    -7.095424
Tpr_TRINITY_DN10844_c2_g7|Tdu_TRINITY_DN25328_c... -0.498146    ...    -4.727989
Tpr_TRINITY_DN15383_c4_g16|Tdu_TRINITY_DN14160_...  0.946122    ...    -3.370146
Tpr_TRINITY_DN1451_c0_g1|Tdu_TRINITY_DN16990_c3_g1 -0.602890    ...    -3.756732
Tpr_TRINITY_DN10853_c1_g4|Tdu_TRINITY_DN21860_c...  0.015868    ...    -7.296158

[5 rows x 6 columns]

In [5]:
df.shape
# shows the number of rows and columns, separately

(8021, 6)

In [6]:
unbias = pd.read_csv("bayes_flag_sig_Filtered_Tdu-Tpr_overlap.V2.csv", names = ['orthologs_no_mapping_bias'])
# this csv do not have header line; use "names" function to add a header line here

In [7]:
unbias.head()

orthologs_no_mapping_bias
0  Tpr_TRINITY_DN11283_c3_g1|Tdu_TRINITY_DN25596_...
1  Tpr_TRINITY_DN11284_c2_g19|Tdu_TRINITY_DN20652...
2  Tpr_TRINITY_DN11285_c1_g4|Tdu_TRINITY_DN17239_...
3  Tpr_TRINITY_DN11295_c2_g2|Tdu_TRINITY_DN23322_...
4  Tpr_TRINITY_DN11295_c2_g3|Tdu_TRINITY_DN12723_...

In [8]:
unbias.shape

(5400, 1)

In [9]:
keys = list(unbias.orthologs_no_mapping_bias)
# creating a list containing all elements from the column orthologs_no_mapping_bias

In [10]:
len(keys)
# count the number of elements in the list

5400

In [11]:
keys[:10]
# show the first few elements of the list

['Tpr_TRINITY_DN11283_c3_g1|Tdu_TRINITY_DN25596_c4_g2',
 'Tpr_TRINITY_DN11284_c2_g19|Tdu_TRINITY_DN20652_c0_g3',
 'Tpr_TRINITY_DN11285_c1_g4|Tdu_TRINITY_DN17239_c2_g3',
 'Tpr_TRINITY_DN11295_c2_g2|Tdu_TRINITY_DN23322_c3_g3',
 'Tpr_TRINITY_DN11295_c2_g3|Tdu_TRINITY_DN12723_c0_g1',
 'Tpr_TRINITY_DN11301_c4_g1|Tdu_TRINITY_DN21600_c2_g2',
 'Tpr_TRINITY_DN11304_c2_g1|Tdu_TRINITY_DN19409_c1_g2',
 'Tpr_TRINITY_DN11305_c3_g1|Tdu_TRINITY_DN19745_c1_g3',
 'Tpr_TRINITY_DN11305_c3_g3|Tdu_TRINITY_DN19745_c1_g6',
 'Tpr_TRINITY_DN11307_c1_g1|Tdu_TRINITY_DN18676_c6_g2']

In [12]:
df_filtered = df[df.index.isin(keys)]
# remain the indexes (orthologs names) in df that are shown in the list keys; output is df_filtered

In [13]:
df_filtered.head()

logFC    ...            B
Tpr_TRINITY_DN10844_c2_g7|Tdu_TRINITY_DN25328_c... -0.498146    ...    -4.727989
Tpr_TRINITY_DN15383_c4_g16|Tdu_TRINITY_DN14160_...  0.946122    ...    -3.370146
Tpr_TRINITY_DN1451_c0_g1|Tdu_TRINITY_DN16990_c3_g1 -0.602890    ...    -3.756732
Tpr_TRINITY_DN15495_c0_g2|Tdu_TRINITY_DN17403_c... -0.349150    ...    -6.427812
Tpr_TRINITY_DN11223_c1_g1|Tdu_TRINITY_DN21812_c...  0.722041    ...    -4.884085

[5 rows x 6 columns]

In [14]:
df_filtered.shape

(3732, 6)

# Among the 8,021 orthologs with DE data, 3,732 showed unbiased mapping

In [16]:
df_filtered.to_csv("DE_Tdu_Tpr_filtered.txt")

# Output file: DE_Tdu_Tpr_filtered.txt

In [32]:
Filtered = pd.read_table("DE_Tdu_Tpr_filtered.txt", sep = ",")

In [33]:
Filtered.shape

(3732, 7)

In [34]:
Filtered.head()

Unnamed: 0    ...            B
0  Tpr_TRINITY_DN10844_c2_g7|Tdu_TRINITY_DN25328_...    ...    -4.727989
1  Tpr_TRINITY_DN15383_c4_g16|Tdu_TRINITY_DN14160...    ...    -3.370146
2  Tpr_TRINITY_DN1451_c0_g1|Tdu_TRINITY_DN16990_c...    ...    -3.756732
3  Tpr_TRINITY_DN15495_c0_g2|Tdu_TRINITY_DN17403_...    ...    -6.427812
4  Tpr_TRINITY_DN11223_c1_g1|Tdu_TRINITY_DN21812_...    ...    -4.884085

[5 rows x 7 columns]

# adj.P.Val >= 0.5 means there is no expression difference between Tdu and Tpr; there are 3,212 of them

In [35]:
Equal = Filtered[Filtered["adj.P.Val"] >= 0.05]

In [25]:
Equal.shape

(3212, 7)

In [36]:
Equal.head()

Unnamed: 0    ...            B
0  Tpr_TRINITY_DN10844_c2_g7|Tdu_TRINITY_DN25328_...    ...    -4.727989
1  Tpr_TRINITY_DN15383_c4_g16|Tdu_TRINITY_DN14160...    ...    -3.370146
2  Tpr_TRINITY_DN1451_c0_g1|Tdu_TRINITY_DN16990_c...    ...    -3.756732
3  Tpr_TRINITY_DN15495_c0_g2|Tdu_TRINITY_DN17403_...    ...    -6.427812
4  Tpr_TRINITY_DN11223_c1_g1|Tdu_TRINITY_DN21812_...    ...    -4.884085

[5 rows x 7 columns]

# adj.P.Val < 0.5 means the expression level is significantly different between Tdu and Tpr
# If logFC is larger than 0, Tdu > Tpr; if logFC is smaller than 0, Tdu < Tpr
# 243 orthologs have higher expression in Tdu; 277 orthologs expressed higher in Tpr

In [37]:
Tdu_higher = Filtered[(Filtered["adj.P.Val"] < 0.05) & (Filtered["logFC"] > 0)]

In [38]:
Tdu_higher.shape

(243, 7)

In [39]:
Tdu_higher.head()

Unnamed: 0    ...            B
5   Tpr_TRINITY_DN14551_c1_g11|Tdu_TRINITY_DN21735...    ...     0.069893
6   Tpr_TRINITY_DN13298_c1_g1|Tdu_TRINITY_DN18431_...    ...    -0.073649
16  Tpr_TRINITY_DN10516_c1_g3|Tdu_TRINITY_DN19561_...    ...    -1.617311
27  Tpr_TRINITY_DN12437_c6_g1|Tdu_TRINITY_DN18152_...    ...    -0.266452
64  Tpr_TRINITY_DN12896_c9_g2|Tdu_TRINITY_DN20213_...    ...     0.336569

[5 rows x 7 columns]

In [40]:
Tpr_higher = Filtered[(Filtered["adj.P.Val"] < 0.05) & (Filtered["logFC"] < 0)]

In [41]:
Tpr_higher.shape

(277, 7)

In [42]:
Tpr_higher.head()

Unnamed: 0    ...            B
18  Tpr_TRINITY_DN11462_c1_g3|Tdu_TRINITY_DN18955_...    ...     1.336065
28  Tpr_TRINITY_DN12134_c0_g2|Tdu_TRINITY_DN21624_...    ...    -1.827389
54  Tpr_TRINITY_DN13550_c0_g1|Tdu_TRINITY_DN10834_...    ...     0.234267
78  Tpr_TRINITY_DN14494_c6_g4|Tdu_TRINITY_DN18357_...    ...     3.371324
97  Tpr_TRINITY_DN12124_c1_g1|Tdu_TRINITY_DN20551_...    ...     6.267121

[5 rows x 7 columns]

In [44]:
Equal.to_csv("DE_Tdu_Tpr_filtered_not_sig_loci.txt", index=False)
Tdu_higher.to_csv("DE_Tdu_Tpr_filtered_sig_loci_Tdu_higher.txt", index=False)
Tpr_higher.to_csv("DE_Tdu_Tpr_filtered_sig_loci_Tpr_higher.txt", index=False)